api/v1alpha1/controlplane_types.go

```diff
@@ -31,9 +31,15 @@ type ControlPlaneSpec struct {
 	// tokenExpirationSeconds is the expiration time for generated auth token
 	// +optional
 	// +kubebuilder:default:=31536000
-	TokenExpirationSeconds *int64            `json:"tokenExpirationSeconds,omitempty"`
-	PostCreateHook         *string           `json:"postCreateHook,omitempty"`
-	PostCreateHookVars     map[string]string `json:"postCreateHookVars,omitempty"`
+	TokenExpirationSeconds *int64             `json:"tokenExpirationSeconds,omitempty"`
+	PostCreateHooks        []PostCreateHookUse `json:"postCreateHooks,omitempty"`
```

Comment:
The PR does not allow for backwards compatibility with the old way to define PostCreate hooks.
Please see comment here for suggested roadmap: https://github.com/kubestellar/kubeflex/issues/225#issuecomment-1981400259